Project-16-Quiz-App/src/components/Question/index.jsx

```diff
@@ -0,0 +1,83 @@
+import PropTypes from "prop-types";
+import { useState } from "react";
+import style from "./question.module.scss";
+
+import QuestionTimer from "../QuestionTimer";
+import Answers from "../Answers";
+
+import QUESTIONS from "../../store/questions.js";
+import {
+  SKIP_QUESTION_TIMEOUT,
+  CORRECT_ANSWER_TIMEOUT,
+  CHECKING_ANSWER_TIMEOUT,
+  ANSWER_NOT_SELECTED,
+} from "../../services/constant.js";
+
+function Question({ questionIndex, onSelectAnswer, onSkipAnswer }) {
+  const [answer, setAnswer] = useState({
+    selectedAnswer: "",
+    isCorrect: null,
+  });
+
+  const handleSelectAnswer = (selectedAnswer) => {
+    setAnswer({
+      selectedAnswer,
+      isCorrect: null,
+    });
+
+    setTimeout(() => {
+      const correctAnswer = QUESTIONS[questionIndex].answers[0];
+      setAnswer({
+        selectedAnswer,
+        isCorrect: correctAnswer === selectedAnswer,
+      });
+
+      setTimeout(() => {
+        onSelectAnswer(selectedAnswer);
+      }, CORRECT_ANSWER_TIMEOUT);
+    }, CHECKING_ANSWER_TIMEOUT);
+  };
+
+  const timer = answer.selectedAnswer
+    ? answer.isCorrect !== ANSWER_NOT_SELECTED
+      ? CORRECT_ANSWER_TIMEOUT
+      : CHECKING_ANSWER_TIMEOUT
+    : SKIP_QUESTION_TIMEOUT;
+
+  let answerState = answer.selectedAnswer
+    ? answer.isCorrect !== ANSWER_NOT_SELECTED
+      ? answer.isCorrect
+        ? "correct"
+        : "wrong"
+      : "answered"
+    : "";
```

Comment:
@sknepali The nested ternary operators can be difficult to read. Consider refactoring this into a function.